content/modules/rest/pages/business-logic.adoc

```diff
@@ -1,20 +1,208 @@
+
+== Services API
+
+Let's look at the first approach to expose business logic to the API Client: the Services API.
+
+To use a Spring bean as part of the Jmix Services API, it needs to fulfill the following criteria:
+
+1. The Spring bean needs to implement an Interface (the Service Interface), that exposes the method signatures.
+2. The Spring bean needs to be registered in the `rest-services.xml` configuration file
+
+=== Service Exposure
+
+To fulfill the first requirement of implementing a Java interface, let's look at the following example:
+
+The Service Interface allows to expose the methods via the REST Services API:
+
+[source,java]
+.CalculationService.java
+----
+public interface CalculationService {
+
+    String NAME = "sample_CalculationService"; // <1>
+
+    int calculate(int a, int b); // <2>
+}
+----
+<1> the NAME attribute is used as the Spring component name and referenced later in the `rest-services.xml` configuration file.
+<2> `calculate` is the exposed method through the Service Interface.
+
+The implementation of the Service Interface contains the implementation of the calculation logic:
+
+[source,java]
+.CalculationServiceBean.java
+----
+import org.springframework.stereotype.Service;
+
+@Service(CalculationService.NAME) //<1>
+public class CalculationServiceBean implements CalculationService {
+
+    public int calculate(int a, int b) {
+        return a + b;
+    }
+}
+----
+<1> The `CalculationServiceBean` is registered as the Spring component `sample_CalculationService`.
+
 
 === Service Configuration
 
-TIP: https://doc.cuba-platform.com/restapi-7.2/#rest_api_v2_services_config
+To use the Services API endpoint you need to define the exposed services. This happens via an XML configuration file, normally called `rest-services.xml`. You need to create this new file in your Jmix application under `src/main/resources`. It lists all exposed service methods with information about their parameters.
+
+
+
+[source,xml]
+.rest-services.xml
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<services xmlns="http://jmix.io/schema/rest/services">
+    <service name="sample_CalculationService"> // <1>
+        <method name="calculate"> // <2>
+            <param name="a" type="int"/> // <3>
+            <param name="b" type="int"/>
+        </method>
+    </service>
+</services>
+----
+<1> the service is registered by its Spring component name.
+<2> each exposed method is registered by its name.
+<3> the parameter needs to be described by its name and optionally by its type.
+
+After the file has been created, and the services have been defined, you need to register the `rest-services.xml` configuration in the `application.properties` of your Jmix application:
+
+[source,properties]
+.application.properties
+----
+jmix.rest.servicesConfig = rest/sample/rest-services.xml
+----
+
+Once the Service is exposed through the Services API, it is possible to invoke the service as an API Client.
```

Comment:
probably "it is possible to invoke the service **from** an API client."

Reply:
done